ScnSoft-Education/LeetCodeTraining/ReverseInteger7/ReverseIntegerSolution.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace LeetCodeTraining.ReverseInteger
+{
+    //https://leetcode.com/problems/reverse-integer/description/
+    public class ReverseIntegerSolution
+    {
+        public static int Reverse(int x)
+        {
+            int reversedX = 0;
+            while (x != 0)
+            {
+                int digit = x % 10;
+                x /= 10;
+                if (reversedX > int.MaxValue / 10 || (reversedX == int.MaxValue / 10 && digit > 7))
```

Comment:
Can you please suggest why here and later in the code digit is compared with constants 7, 8?

Reply:
Yes, I had to make research about the int type of variables before solving the problem. In that part of the code, I check whether the result remains within the range of 32-bit integer numbers. If the result would exceed this range after adding a new digit from a given number, the Reverse method will return 0. I compare the current digit with the constants 7 and 8 because the MaxValue of the integer type is 2,147,483,647, which ends with 7, and the MinValue of that type in C# is -2,147,483,648, which ends with 8.

Reply:
Ok, got it, thanks.
You can get this values as int.MaxValue % 10, Math.Abs(int.MinValue % 10), in this case if the size of int will be changed in the future your code still will be working, but for now 7, 8 is fine I guess.